app/controllers/users_controller.rb

```diff
@@ -37,6 +37,8 @@ def json
     @user = User.find(params[:id])
     authorize @user
 
+    @user.current_activity = Activity.find(params[:activity_id])
```

Comment:
We need to set the current activity here, since `can_order` will otherwise be set to `false` if the user has negative credit, regardless of previous orders on this activity